src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,553 @@
+* `measure_theory.measure.singular_part` : If a pair of measures `have_lebesgue_decomposition`,
+  then `singular_part` chooses the measure from `have_lebesgue_decomposition`, otherwise it
+  returns the zero measure.
+* ``measure_theory.measure.radon_nikodym_deriv` : If a pair of measures
+  `have_lebesgue_decomposition`, then `radon_nikodym_deriv` chooses the measurable function from
+  `have_lebesgue_decomposition`, otherwise it returns the zero function.
+
+## Main results
+
+* `measure_theory.measure.have_lebesgue_decomposition_of_finite_measure` :
+  the Lebesgue decomposition theorem.
+
+## To do
+
+The Lebesgue decomposition theorem can be generalized to σ-finite measures from the finite version.
+
+# Tags
+
+Lebesgue decomposition theorem
+-/
+
+noncomputable theory
+open_locale classical measure_theory nnreal ennreal
+
+variables {α β : Type*} [measurable_space α]
+
+namespace measure_theory
+
+namespace measure
+
+/-- A pair of measures `μ` and `ν` is said to `have_lebesgue_decomposition` if there exists a
+measure `ξ` and a measurable function `f`, such that `ξ` is mutually singular with respect to
+`ν` and `μ = ξ + ν.with_density f`. -/
+def have_lebesgue_decomposition (μ ν : measure α) : Prop :=
+∃ (p : measure α × (α → ℝ≥0∞)), measurable p.2 ∧ p.1 ⊥ₘ ν ∧ μ = p.1 + ν.with_density p.2
+
+/-- If a pair of measures `have_lebesgue_decomposition`, then `singular_part` chooses the
+measure from `have_lebesgue_decomposition`, otherwise it returns the zero measure. -/
+@[irreducible]
+def singular_part (μ ν : measure α) : measure α :=
+if h : have_lebesgue_decomposition μ ν then (classical.some h).1 else 0
+
+/-- If a pair of measures `have_lebesgue_decomposition`, then `radon_nikodym_deriv` chooses the
+measurable function from `have_lebesgue_decomposition`, otherwise it returns the zero function. -/
+@[irreducible]
+def radon_nikodym_deriv (μ ν : measure α) : α → ℝ≥0∞ :=
+if h : have_lebesgue_decomposition μ ν then (classical.some h).2 else 0
+
+lemma have_lebesgue_decomposition_spec {μ ν : measure α} (h : have_lebesgue_decomposition μ ν) :
+  measurable (radon_nikodym_deriv μ ν) ∧ (singular_part μ ν) ⊥ₘ ν ∧
+  μ = (singular_part μ ν) + ν.with_density (radon_nikodym_deriv μ ν) :=
+begin
+  rw [singular_part, radon_nikodym_deriv, dif_pos h, dif_pos h],
+  exact classical.some_spec h,
+end
+
+@[measurability]
+lemma measurable_radon_nikodym_deriv (μ ν : measure α) :
+  measurable $ radon_nikodym_deriv μ ν :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { exact (have_lebesgue_decomposition_spec h).1 },
+  { rw [radon_nikodym_deriv, dif_neg h],
+    exact measurable_zero }
+end
+
+lemma mutually_singular_singular_part (μ ν : measure α) :
+  singular_part μ ν ⊥ₘ ν :=
+begin
+  by_cases h : have_lebesgue_decomposition μ ν,
+  { exact (have_lebesgue_decomposition_spec h).2.1 },
+  { rw [singular_part, dif_neg h],
+    exact mutually_singular.zero.symm }
+end
+
```

Comment:
can you also add a lemma `have_lebesgue_decomposition.singular_part_add_eq_deriv` saying that, assuming `have_lebesgue_decomposition μ ν`, one has `(singular_part μ ν) + ν.with_density (radon_nikodym_deriv μ ν) = μ`? I know that this is already available in `have_lebesgue_decomposition_spec`, but this is the only part of this lemma which is not always true, and the most important one, so making it available through dot notation sounds like a good idea.